pkg/registry/core/pod/etcd/eviction.go

```diff
@@ -116,6 +116,10 @@ func (r *EvictionREST) Create(ctx api.Context, obj runtime.Object) (runtime.Obje
 }
 
 func (r *EvictionREST) checkAndDecrement(namespace string, podName string, pdb policy.PodDisruptionBudget) (ok bool, err error) {
+	if pdb.Status.ObservedGeneration == nil || *pdb.Status.ObservedGeneration != pdb.Generation {
+		return false, fmt.Errorf("pdb status has not been updated after spec change")
```

Comment:
I'm wondering if this should be an error. Because this is the situation that will be happening in a real system. It just means that we didn't yet updated PDB. So I think that we should:
```
return false, nil
```
here.

For me error is for something that is not expected, and this can definitely happen.